bmorph/evaluation/StatCor_Vis.py

```diff
@@ -0,0 +1,566 @@
+import numpy as np
+import xarray as xr
+import pandas as pd
+import seaborn as sns
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import sys
+import bmorph
+
+sns.set_context('talk')
+mpl.style.use('seaborn-bright')
+mpl.rcParams['figure.figsize'] = (12, 9)
+CFS_TO_CMS = 35.314666212661
+
+import descStats as dst
+import format_data as fd;
+
+colors99p99=['Maroon','Navy','Black','Orange','Yellow','Blue','Grey','Pink','Lavender']
+siteList = fd.sites
+
+def custom_legend(names: list(),colors=colors99p99):
+    """
+    Custom Legend
+        creates a list of patches to be passed in as
+        `handels` for the plt.legends function
+    ----
+    names: a list of legend names
+    colors: a list of colors
+    
+    It is assumed that the order of colors and corresponding
+    names for plotted values match already
+    """
+    legend_elements = list()
+    for i,name in enumerate(names):
+        legend_elements.append(mpl.patches.Patch(facecolor=colors[i],label=name))
+    return legend_elements
+
+def calc_water_year(df: pd.DataFrame):
+    """
+    Calculate Hydrologic Year
+    ----
+    
+    df: pandas DataFrame with a DataTime index
+    
+    Returns: an index grouped by Hydrologic Year
+    
+    """
+    return df.index.year + (df.index.month >= 10).astype(int)
+
+def find_index_hyear(data: pd.DataFrame) -> np.int:
```

Comment:
I'm a little unclear on how this function should be used. Is the goal to find the first index where the month is October? You can do that a little bit more simply with:
```
def find_index_hyear(df: pd.DataFrame) -> np.int
    return np.argwhere(df.index.month == 10)[0][0]
```

No need to do anything for this, just a heads up.

Reply:
Yeah, that's all that was for, thanks for the note